domain/schema/model/sql/0022-resource.sql

```diff
@@ -29,103 +29,85 @@ INSERT INTO resource_state VALUES
 
 CREATE TABLE resource (
     uuid TEXT NOT NULL PRIMARY KEY,
-    application_uuid TEXT NOT NULL,
+    charm_uuid TEXT NOT NULL,
     name TEXT NOT NULL,
     revision INT,
     origin_type_id INT NOT NULL,
     state_id INT NOT NULL,
-    size INT,
-    hash TEXT,
-    hash_type_id TEXT,
     created_at TIMESTAMP NOT NULL,
-    CONSTRAINT fk_resource_application_uuid
-    FOREIGN KEY (application_uuid)
-    REFERENCES application (uuid),
+    -- last_polled is when the repository was last polled for new resource
+    -- revisions. Only set if resource_state is 1.
+    last_polled TIMESTAMP,
+    CONSTRAINT fk_charm
+    FOREIGN KEY (charm_uuid)
+    REFERENCES charm (uuid),
     CONSTRAINT fk_resource_name
     FOREIGN KEY (name)
-    REFERENCES resource_meta (name),
+    REFERENCES charm_resource (name),
```

Comment:
**suggestion(maybe_issue):**  I think it would be better to FK both `name` and `charm_uuid` to the same table (charm_uuid), for the sake of clarity. resource shouldn't have any direct link to charm.

**suggestion:** Even if [sqlite may have surprising behavior, yet perfectly ok once we now with compound fk](https://stackoverflow.com/questions/19656431/trouble-with-composite-foreign-keys-in-sqlite3), I think it would be better to use a compound foreign key here to enforce my point above:

```SQL
 FOREIGN KEY (charm_uuid, name) 
 REFERENCES charm_resource (charm_uuid, name)
```

Reply:
The `charm_uuid` key is removed. The link is implied by that to `charm_resource`.